common/version/semver_test.go

```diff
@@ -0,0 +1,175 @@
+package version
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func verifySemver(t *testing.T, str string, major int, minor int, patch int, errata string) {
+	semver, err := SemverFromString(str)
+	require.NoError(t, err)
+	require.Equal(t, major, semver.Major())
+	require.Equal(t, minor, semver.Minor())
+	require.Equal(t, patch, semver.Patch())
+	require.Equal(t, errata, semver.Errata())
+	require.Equal(t, str, semver.String())
+}
+
+func TestSerialization(t *testing.T) {
```

Comment:
Looks like you're testing eratta, but not the "extra" style which is defined within `semver`